### DISCUSSION TITLE
Elastalert ignores the documents that are already processed/seen in previous overlapping query

### DISCUSSION THREAD

**Discussion by RAJMODI038** [jertel/elastalert2#1053] [Q&A] [answered]
From elastalert documentation https://elastalert2.readthedocs.io/en/latest/running_elastalert.html#id1, I found that by default elastalert ignores the documents that are already processed/seen in previous overlapping query.

**'already seen' refers to documents that were already counted in a previous overlapping query and will be ignored**

![image](https://user-images.githubusercontent.com/40465877/208103551-e689d9d0-fbf2-4623-9a6a-26b20933bd91.png)

Is there any way to override this functionality? or can someone point out what part of the code filters out these documents?
stackoverflow: https://stackoverflow.com/questions/74824190/elastalert-ignores-the-documents-that-are-already-processed-seen-in-previous-ove

**Comment by jertel** [comment, marked as answer]
You can use `scan_entire_timeframe` to force ElastAlert 2 to use the entire `timeframe` duration in the search. This only applies to the first search after ElastAlert 2 restarts. Subsequent searches will not allow queries to overlap as this would cause duplicate alerts to fire. See the following code for more specific logic on this: 

https://github.com/jertel/elastalert2/blob/ebd467be4d3503ff7faf62b504a729b3ce7908f6/elastalert/elastalert.py#L635
